lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -52,25 +52,22 @@
             - Valid azure location. Defaults to location of the resource group.
         default: resource_group location
         required: false
-    virtual_network_resource_group:
```

Comment:
will it cause existing user using this parameter break by delete? how about keep as alias?

Reply:
this would have been new added in 2.6, current user is using devel version, I think we can have this 'breaking change' in devel version

Reply:
is it necessary to introduce the break? I would suggest to keep virtual_network_resource_group. If remove this argument, user will not able to create a NIC with vnet in another resource group.

Reply:
This is an agreement we reached in a sync up meeting. Refer resource in other resource group support 3 methods:

* accept resource name, and the resource is in the same resource group
* accept a resource id
* accept a sub dict, which contains resource group and name

Here I remove this option to support the 3 methods. Since this option's `version_added` is `2.6`. That means no user is using this option because this hasn't been released.

Reply:
The aliases section means both `virtual_network` or `virtual_network_name` can be used when specifying this option. The name change is to bring it more in line with other modules, plus now it can take a dict with more than just the name